addons/stock_landed_costs/models/stock_landed_cost.py

```diff
@@ -267,15 +267,15 @@ def compute_landed_cost(self):
                             value = (line.price_unit / total_line)
 
                         if rounding:
-                            value = tools.float_round(value, precision_rounding=rounding, rounding_method='UP')
-                            fnc = min if line.price_unit > 0 else max
-                            value = fnc(value, line.price_unit - value_split)
+                            value = tools.float_round(value, precision_rounding=rounding, rounding_method='HALF-UP')
                             value_split += value
 
                         if valuation.id not in towrite_dict:
                             towrite_dict[valuation.id] = value
                         else:
                             towrite_dict[valuation.id] += value
+                if (r_diff := self.currency_id.round(line.price_unit - value_split)) != 0:
```

Comment:
```suggestion
                rounding_diff = self.currency_id.round(line.price_unit - value_split)
                if not float_is_zero(rounding_diff, precision_rounding=rounding):
```
Even after a `round()`, there is no guarantee that a zero-like value would be actually `0`.